beater/telemetry.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beater
+
+import (
+	"github.com/elastic/apm-server/beater/config"
+	"github.com/elastic/apm-server/idxmgmt"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/idxmgmt/ilm"
+	"github.com/elastic/beats/v7/libbeat/monitoring"
+)
+
+var apmRegistry = monitoring.GetNamespace("state").GetRegistry().NewRegistry("apm-server")
+var configRegistry = apmRegistry.NewRegistry("configuration")
+
+type configTelemetry struct {
+	rumEnabled                *monitoring.Int
+	apiKeysEnabled            *monitoring.Int
+	kibanaEnabled             *monitoring.Int
+	pipelinesEnabled          *monitoring.Int
+	pipelinesOverwrite        *monitoring.Int
+	setupTemplateEnabled      *monitoring.Int
+	setupTemplateOverwrite    *monitoring.Int
+	setupTemplateAppendFields *monitoring.Int
+	ilmEnabled                *monitoring.Int
+	ilmSetupEnabled           *monitoring.Int
+	ilmSetupOverwrite         *monitoring.Int
+	ilmSetupRequirePolicy     *monitoring.Int
+	jaegerGRPCEnabled         *monitoring.Int
+	jaegerHTTPEnabled         *monitoring.Int
+	sslEnabled                *monitoring.Int
+}
+
+var configMonitors = &configTelemetry{
+	rumEnabled:                monitoring.NewInt(configRegistry, "rumEnabled"),
```

Comment:
What's the reason for diverging from the config names we use? Why `rumEnabled` and not `rum.enabled`?

Reply:
Well, Ill argue that this is using the config names. `camelCase` instead of `dots` to avoid mapping to object in ES.

Reply:
> to avoid mapping to object in ES

... but why? Case in point: `management.enabled`